File: system-test/dotnet/SystemTests/Legacy/Mod06/SmokeTests/System/ShopSmokeTest.cs
using Common;
using SystemTests.Legacy.Mod06.Base;
using Dsl.Core.Shop;
using Optivem.Testing;
using Shouldly;
using Xunit;

namespace SystemTests.Legacy.Mod06.SmokeTests.System;

public class ShopSmokeTest : BaseChannelDriverTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldBeAbleToGoToShop(Channel channel)
    {
        await SetChannelAsync(channel);

        var result = await _shopDriver!.GoToShopAsync();
        result.ShouldBeSuccess();
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod06/smoke/system/ShopSmokeTest.java
package com.optivem.shop.systemtest.legacy.mod06.smoke.system;

import com.optivem.shop.systemtest.legacy.mod06.base.BaseChannelDriverTest;
import com.optivem.shop.dsl.channel.ChannelType;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

import static com.optivem.shop.dsl.common.ResultAssert.assertThatResult;

class ShopSmokeTest extends BaseChannelDriverTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldBeAbleToGoToShop() {
        var result = shopDriver.goToShop();
        assertThatResult(result).isSuccess();
    }
}

